icache.go

```diff
@@ -10,10 +11,28 @@ type Cache interface {
 	Del(key string) error
 
 	BatchGet(keys ...string) ([][]byte, error)
+	// BatchSet
 	// key1, value1, key2, value2 ...
 	// string, bytes, string,bytes
 	BatchSet(keyvalues ...interface{}) error
 	Close() error
 }
 
-var ErrCacheMiss = errors.New("not found in cache")
+type CacheCtx interface {
+	SetCtx(ctx context.Context, key string, value []byte) error
```

Comment:
another concern is value type here is a bit restrictive, for example, Redis would accept any `interface{}` as long as it knows how to marshall it, for example, implement `encoding.BinaryMarshaler`

and although less important libMemcache can even use structs as keys 🤔